### PR TITLE
NAS-120508 / 23.10 / WebUI validation of VDEV redundancy across VDEV types

### PR DESCRIPTION
**Pull request by AlexKarpov98**
code review is enough.
Checked together with Chris Peredun.

### CODE REVIEW

**Review by dszidi — CHANGES_REQUESTED** [review]
Please update src/app/services/storage.service.spec.ts file to reflect affected scenarios and desired outcomes.

File: src/app/pages/storage/components/dashboard-pool/topology-card/topology-card.component.spec.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ describe('TopologyCardComponent', () => {
 
     // Redundancy level should match data VDEVs
     expect(captions[1]).toHaveText('Metadata');
-    expect(values[1]).toHaveText('Redundancy Mismatch');
+    expect(values[1]).toHaveText('1 x MIRROR | 3 wide | 8 TiB');
```
Comment (dszidi) [inline src/app/pages/storage/components/dashboard-pool/topology-card/topology-card.component.spec.ts:88]: For this test let's have both redundancy mismatched and matched scenarios. Both Metadata and Dedup need to match redundancy level of data vdevs. Perhaps leave the metadata vdevs as you already have it but add dedup topology category to mockstorage and deliberately mismatch it? That way we can test for the error message.

**Review by dszidi — CHANGES_REQUESTED** [review]
Requested adjustment to topology card spec file

File: src/app/core/testing/utils/mock-storage-generator.utils.ts [hunk 2]
```diff
@@ -124,9 +124,9 @@ export class MockStorageGenerator {
         break;
       }
       case MockStorageScenario.Uniform: {
-        if (options.layout === TopologyItemType.Stripe || options.layout === TopologyItemType.Disk) {
-          options.layout = TopologyItemType.Mirror;
-        }
+        // if (options.layout === TopologyItemType.Stripe || options.layout === TopologyItemType.Disk) {
```
[the remark this answers is not in the document]
Comment (AlexKarpov98) [reply src/app/core/testing/utils/mock-storage-generator.utils.ts:127]: my mistake

**Comment by AlexKarpov98** [comment]
@dszidi - please check it now :)

**Review by dszidi — APPROVED** [review]
LGTM

**Pull request merged**